test/sql/date/test_timestamp.cpp

```diff
@@ -134,3 +134,60 @@ TEST_CASE("Test storage for timestamp type", "[timestamp]") {
 	}
 	DeleteDatabase(storage_database);
 }
+
+TEST_CASE("Test timestamp functions", "[timestamp]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	result = con.Query("SELECT AGE(TIMESTAMP '1957-06-13');");
+	auto current_timestamp = Timestamp::GetCurrentTimestamp();
+	auto interval = Timestamp::GetDifference(Timestamp::FromString("1957-06-13"), current_timestamp);
+	auto timestamp = Timestamp::IntervalToTimestamp(interval);
+	auto years = timestamp.year;
+	auto months = timestamp.month;
+	auto days = timestamp.day;
+
+	auto output = std::to_string(years);
+	output += " years ";
+	output += std::to_string(months);
+	output += " mons ";
+	output += std::to_string(days);
+	output += " days";
+	REQUIRE(CHECK_COLUMN(result, 0, {output.c_str()}));
+
+	result = con.Query("SELECT AGE(TIMESTAMP '2001-04-10', TIMESTAMP '1957-06-13');");
+	REQUIRE(CHECK_COLUMN(result, 0, {"43 years 9 mons 27 days"}));
+
+	result = con.Query("SELECT age(TIMESTAMP '2014-04-25', TIMESTAMP '2014-04-17');");
+	REQUIRE(CHECK_COLUMN(result, 0, {"8 days"}));
+
+	result = con.Query("SELECT age(TIMESTAMP '2014-04-25', TIMESTAMP '2014-01-01');");
+	REQUIRE(CHECK_COLUMN(result, 0, {"3 mons 24 days"}));
+
+	result = con.Query("SELECT age(TIMESTAMP '2019-06-11', TIMESTAMP '2019-06-11');");
+	REQUIRE(CHECK_COLUMN(result, 0, {"00:00:00"}));
+
+	result = con.Query(" SELECT age(timestamp '2019-06-11 12:00:00', timestamp '2019-07-11 11:00:00');");
+	REQUIRE(CHECK_COLUMN(result, 0, {"-29 days -23:00:00"}));
+
+	REQUIRE_FAIL(con.Query("SELECT AGE(NULL, NULL);"));
+
+	REQUIRE_FAIL(con.Query("SELECT AGE(TIMESTAMP '1957-06-13', NULL);"));
+
+	// create and insert into table
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE timestamp(t1 TIMESTAMP, t2 TIMESTAMP)"));
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO timestamp VALUES('2001-04-10', '1957-06-13')"));
+
+	result = con.Query("SELECT AGE(t1, TIMESTAMP '1957-06-13') FROM timestamp;");
+	REQUIRE(CHECK_COLUMN(result, 0, {"43 years 9 mons 27 days"}));
+
+	result = con.Query("SELECT AGE(TIMESTAMP '2001-04-10', t2) FROM timestamp;");
+	REQUIRE(CHECK_COLUMN(result, 0, {"43 years 9 mons 27 days"}));
+
+	result = con.Query("SELECT AGE(t1, t2) FROM timestamp;");
+	REQUIRE(CHECK_COLUMN(result, 0, {"43 years 9 mons 27 days"}));
+
+	result = con.Query("SELECT AGE(t1, t2) FROM timestamp WHERE t1 > '1957-12-12';");
```

Comment:
Also here make sure the filter actually removes some, but not all, rows (e.g. if your table has 3 rows, make sure 1-2 rows pass the filter).